app/ide-desktop/lib/content/src/index.ts

```diff
@@ -119,104 +119,132 @@ function displayDeprecatedVersionDialog() {
 }
 
 // ========================
-// === Main Entry Point ===
+// === Main entry point ===
 // ========================
 
 interface StringConfig {
     [key: string]: StringConfig | string
 }
 
-class Main {
-    async main(inputConfig: StringConfig) {
-        const config = Object.assign(
-            {
-                loader: {
-                    wasmUrl: 'pkg-opt.wasm',
-                    jsUrl: 'pkg.js',
-                    assetsUrl: 'dynamic-assets',
-                },
-            },
-            inputConfig
-        )
-
-        const appInstance = new app.App({
-            config,
-            configOptions: contentConfig.OPTIONS,
-            packageInfo: {
-                version: BUILD_INFO.version,
-                engineVersion: BUILD_INFO.engineVersion,
+// Hack to mutate `configOptions.OPTIONS`
```

Comment:
Why we need to hack it? Please add more description and fixme issue how it should be solved

Reply:
not sure if there's a way to solve this.
the issue is that `new app.App` is what mutates `configOptions.OPTIONS` so it *must* be executed before checking whether we should run auth:
```ts
if (
    (contentConfig.OPTIONS.options.authentication.value ||
        contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value) &&
    contentConfig.OPTIONS.groups.startup.options.entry.value ===
        contentConfig.OPTIONS.groups.startup.options.entry.default
) {
```